File: LeetCode/Explore/TopInterview/Amazon/DynamicalProgramming/121_best_time_to_buy_and_sell_stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minCost = prices[0]
        maxGain = 0
        
        for i in range(1, len(prices), 1):
          if prices[i] < minCost:
            minCost = prices[i]
          else:
            maxGain = max(maxGain, prices[i] - minCost)
            
        return maxGain

File: LeetCode/Explore/TopInterview/Amazon/DynamicalProgramming/121_best_time_to_buy_and_sell_stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
      int minCost = prices[0], maxGain = 0;
      for (int i = 1; i < prices.size(); i++) {
        if (prices[i] < minCost) {
          minCost = prices[i];
        }
        else {
          maxGain = max(maxGain, prices[i] - minCost);
        }
      }
      return maxGain;
    }
};
